src/error.rs

```diff
@@ -98,6 +98,40 @@ impl Error {
     }
 }
 
+impl Error {
+    pub(crate) fn into_infallible_response(self) -> Response {
+        use http::header::HeaderValue;
+
+        let status_code = self.status;
+        let message = self.to_string();
+
+        match self.into_response() {
+            Ok(response) => response,
+            Err(error) => {
+                let mut response = Response::new();
+
+                response.headers_mut().insert(
+                    http::header::CONTENT_TYPE,
+                    HeaderValue::from_static("text/plain; charset=utf-8"),
+                );
+
+                if let Some(length) = HeaderValue::from_str(&message.len().to_string()).ok() {
+                    response
+                        .headers_mut()
+                        .insert(http::header::CONTENT_LENGTH, length);
+                }
+
+                *response.status_mut() = status_code;
+                *response.body_mut() = message.into();
+
+                eprintln!("Failed to convert error into response: {}", error);
```

Comment:
I won't leave this eprintln here permanently. I'll refine the code in this part of the match branch, adding comments and finding a better way to indicate why the error couldn't be converted into a response. Maybe an error callback on app is our best option for now?